File: frontend/Modele/Carte.js
import { Couleur } from "./Couleur.js";
import { Mot } from "./Mot.js";
class Carte {
    #mot;
    #couleur;
    #face_cachee;
    #ligne;
    #colonne;

    constructor(){
        this.#mot = new Mot();
        this.#couleur = new Couleur();
        this.#face_cachee = true;
        this.#ligne = 0;
        this.#colonne = 0;
    }
    hydrate(data){
        this.#mot.hydrate(data.mot);
        this.#couleur.hydrate();
        this.#face_cachee = data.face_cachee;
        this.#ligne = data.ligne;
        this.#colonne = data.colonne;
    }
}

File: backend/src/modele/Carte.java
package modele;

public class Carte {
    private Mot mot;
    private String code_partie;
    private int couleur;
    private Boolean face_cachee;
    
    public Carte(Mot mot, String code_partie, int couleur, Boolean face_cachee) {
        this.mot = mot;
        this.code_partie = code_partie;
        this.couleur = couleur;
        this.face_cachee = face_cachee;
    }

    public Mot getMot() {
        return mot;
    }

    public String getCode_partie() {
        return code_partie;
    }

    public int getCouleur() {
        return couleur;
    }

    public Boolean getFace_cachee() {
        return face_cachee;
    }

}
